timApp/util/flask/search.py

```diff
@@ -272,11 +272,11 @@ def get_search_match_count(results: list[SearchResult], rt: Type[SearchResult])
     :return: total number of matches found for the search
     """
     if results and not type(results[0]) is rt:
-        results = list(result for result in results if result.__class__ is rt)
+        results = list(result for result in results if isinstance(result, rt))
```

Comment:
```suggestion
        results = [result for result in results if isinstance(result, rt)]
```

Reply:
Muutamassa muussa kohdassa näyttää olevan sama juttu, eli pythonimaisempi olisi vaihtaa `list(a for a in b)` muotoon `[a for a in b]`.